numpy/_core/src/umath/ufunc_object.c

```diff
@@ -806,8 +807,8 @@ check_for_trivial_loop(PyArrayMethodObject *ufuncimpl,
                  * can  force cast to bool)
                  */
             }
-            else if (PyArray_MinCastSafety(safety, casting) != casting) {
-                return 0;  /* the cast is not safe enough */
+            else if (is_safe != 1) {
```

Comment:
Just `!is_safe`?

Reply:
This one actually needs to specifically compare with 1, because `is_safe` can be negative here without the error indicator being set above.

Worth adding a comment you think?

Reply:
specifically `test_trivial_loop_invalid_cast` fails if I apply your suggestion

Reply:
Ah, yes, I missed that the constraint above also checked the error, sorry.